src/content/docs/kubernetes-pixie/kubernetes-integration/get-started/kubernetes-components.mdx

```diff
@@ -0,0 +1,129 @@
+
+A specific `nrk8s-ksm` deployment uses an endpoint informer to locate the IP of the KSM pod and scrape it. The informer caches the list of informers in the cluster locally and watch for new ones.
+
+See [Bring your own KSM](https://github.com/newrelic/helm-charts/blob/master/charts/nri-bundle/README.md#bring-your-own-ksm) for more information about KSM supported versions.
+
+<Callout variant="important">
+  Cloud provider host tags, like `aws.ec2.*`, are no longer included in samples related to entities that are not strictly tied to a particular node: `K8sNamespaceSample`, `K8sDeploymentSample`, `K8sReplicasetSample`, `K8sDaemonsetSample`, `K8sStatefulsetSample`, `K8sServiceSample`, and `K8sHpaSample`.
+  If you need to add an attribute to these samples, use `customAttributes`.
+</Callout>
+
+## Kubelet component [#nrk8s-kubelet]
+
+The Kubelet is the “Kubernetes agent”, a service that runs on every Kubernetes node. It's responsible for creating the containers as instructed by the control plane. Since it's the Kubelet who partners closely with the Container Runtime, it's the main source of infrastructure metrics for our integration, such as use of CPU, memory, disk, network, etc.
+
+`nrk8s-kubelet` runs as a DaemonSet where each instance gathers metric from the Kubelet running in the same node as it is. This component connects to the Kubelet using the Node IP. If this process fails, `nrk8s-kubelet` will fall back to reach the node through the API Server proxy. 
+
+<Callout variant="important">
+  If you have very large clusters, proxying many kubelets might increase the load in the API server.
+</Callout>
+
+To know if the API Server is being used as a proxy, check the logs and look for a similar message to this one:
+
+```
+Trying to connect to kubelet through API proxy
+```
+
+## Control plane component [#nrk8s-controlplane]
+
+New Relic deploys `nrk8s-controlplane` as a DaemonSet with `hostNetwork: true`. The configuration is structured to support autodiscover and static endpoints.
+A wide range of known defaults are provided as configuration entries, which allows you to tweak autodiscovery to your needs.
+
+You can have multiple endpoints per selector and add a probe mechanism, which automatically detects the correct one. You can use the same selector to try different configurations such as ports or protocols.
+
+Scraping configuration for the `etcd CP` component looks like the following where the same structure and features applies for all components:
+
+```yaml
+config:
+  etcd:
+    enabled: true
+    autodiscover:
+      - selector: "tier=control-plane,component=etcd"
+        namespace: kube-system
+        matchNode: true
+        endpoints:
+          - url: https://localhost:4001
+            insecureSkipVerify: true
+            auth:
+              type: bearer
+          - url: http://localhost:2381
+    staticEndpoint:
+      url: https://url:port
+      insecureSkipVerify: true
+      auth: {}
```

Comment:
This snippet in isolation will likely confuse users. "Where do I need to copypaste this?"
The answer is in a very specific section of the `values.yaml` file, indented under several keys that are not mentioned in this page. `values.yaml` file is part of Helm, which also has not been introduced yet when the user reads this.